### ISSUE TITLE
TLN_SetSpritePicture Strange Behaviour

### ISSUE THREAD

**Issue by acheddadi** [megamarc/Tilengine#124]
Hello,

I seem to be running into an issue where passing any non-zero value for the entry parameter in the method `TLN_SetSpritePicture`, results in multiple sprites (from my sprite set) being drawn on screen. Here are is a screenshot of the issue:
![Screenshot_2024-12-12_18-50-53](https://github.com/user-attachments/assets/abefe442-96ef-420e-a585-2beab0e5b654)
The above screenshot shows the "Pawn" chess piece as being properly drawn, however, the "Rook" and the "Knight" both show additional sprites on either side of them.

Here is my spriteset as well as my mapping:
![chess-pieces](https://github.com/user-attachments/assets/abe24fb8-8d6a-40d4-bc68-3b6bc33cb98b)

> chess-pieces0,0,0,64,64
> chess-pieces1,64,0,128,64
> chess-pieces2,128,0,192,64
> chess-pieces3,192,0,256,64
> chess-pieces4,256,0,320,64
> chess-pieces5,320,0,384,64

Setting the index to 0 will correctly show the first sprite in the sprite set, however any other value results in a strange offset where a multitude of sprites are drawn at once. Here is a snippet of my code where SetSpritePicture is being called:
```
struct PaintRequest
{
    public:
        int spriteID;
        int frame;
        int layer;
        Vector3 position;
        Vector3 pivot;
        Vector3 scale;
        PaintRequest* nextPaintRequest;
};

void Painter::Paint(PaintRequest drawCall)
{
    TLN_SetFirstSprite(drawCall.spriteID);

    TLN_SetSpritePivot(drawCall.spriteID, drawCall.pivot.x, drawCall.pivot.y);

    TLN_SetSpritePicture(drawCall.spriteID, drawCall.frame);

    TLN_SetSpritePosition(
        drawCall.spriteID,
        drawCall.position.x,
        drawCall.position.y);

    if (drawCall.scale.x != 0.0f && drawCall.scale.y != 0.0f)
    {
        TLN_SetSpriteScaling(
            drawCall.spriteID,
            drawCall.scale.x >= 0.0f ? drawCall.scale.x : -drawCall.scale.x,
            drawCall.scale.y >= 0.0f ? drawCall.scale.y : -drawCall.scale.y);
    }
}
```

Thank you in advance for you help!

**Comment by acheddadi** [comment]
I took a closer look at some of the included examples and realized that my mapping was incorrect. I was under the impression that the format for each entry should be: "name, topLeftCornerX, topLeftCornerY, bottomRightCornerX, bottomRightCornerY", when actually the expected format is "name, topLeftCornerX, topLeftCornerY, width, height".

Sorry for the mix up!

**Comment by megamarc** [comment]
Thanks, I was going to take a look at it but you found the cause.Happy tilengine and let me know if you face some issueEl 13 dic. 2024 18:15, acheddadi ***@***.***> escribió:
Closed #124 as completed.
